src/urls/util.ts

```diff
@@ -14,32 +14,14 @@
 
 import {posix as path} from 'path';
 import {Document} from 'polymer-analyzer';
+import {dependencyMap} from '../manifest-converter';
 
-import {dependencyMap} from './manifest-converter';
-
-const htmlExtension = '.html';
-
-/**
- * A URL path to an document, pre-conversion. Always relative to the current
- * project (package, workspace, etc).
- */
-export type OriginalDocumentUrl = string&{_OriginalDocumentUrl: never};
-
-/**
- * A URL path to a document, post-conversion. Uses npm naming for all urls
- * containing package names. Always relative to the current project (package,
- * workspace, etc).
- */
-export type ConvertedDocumentUrl = string&{_ConvertedDocumentUrl: never};
-/**
- * A file path to where a document will be written, post-conversion. Unlike
- * ConvertedDocumentUrl, this url keeps the original package folder and Bower
- * package name. Useful in workspace projects where converted files should still
- * be written to the original analyzed repo.
- */
-export type ConvertedDocumentFilePath =
-    string&{_ConvertedDocumentFilePath: never};
+import {ConvertedDocumentUrl, OriginalDocumentUrl} from './types';
 
+/** The HTML file extension. */
+export const htmlExtension = '.html';
```

Comment:
Do these make sense as constants? Is there any case where we'd expect them to be typo'd or that it would be clearer to have the name of the constant rather than just the string literal?

Reply:
That's a fair point, thanks for bringing it up. I don't feel too strongly either way here, but the one place that it definitely feels safer is here:

```ts
if (jsUrl.endsWith(htmlExtension)) {
  jsUrl = fixHtmlExtension(jsUrl) as ConvertedDocumentUrl;
}
```

Checking for the existence of a constant (`htmlExtension`) makes me feel more confident calling the method built to handle that constant (`fixHtmlExtension()`). But digging deeper, that's really the result of `fixHtmlExtension()` breaking a non-html URL. Let me move the check in there, rename it `rewriteHtmlExtensionIfFound(str)`, and then the constants are no longer needed!